From b9475d316fdd67bd5e53fdf3abb30507bc06fd2f Mon Sep 17 00:00:00 2001
From: avimn <avimn@amd.com>
Date: Fri, 3 Mar 2017 09:07:48 +0530
Subject: [PATCH 2060/2094] drm/amdgpu: Disable CG clockgating on stoney

Coarse grain clockgating is causing IOMMU errors and filesystem corruption
on stoney embedded boards. Disabling CGCG is a temporary workaround until the
issue is fixed properly.

Signed-off-by: Avinash M N <avimn@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/vi.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/vi.c b/drivers/gpu/drm/amd/amdgpu/vi.c
index 48fb373..35ebd54 100644
--- a/drivers/gpu/drm/amd/amdgpu/vi.c
+++ b/drivers/gpu/drm/amd/amdgpu/vi.c
@@ -1046,7 +1046,11 @@ static int vi_common_early_init(void *handle)
 			AMD_CG_SUPPORT_GFX_CP_LS |
 			AMD_CG_SUPPORT_GFX_CGTS |
 			AMD_CG_SUPPORT_GFX_CGTS_LS |
-			AMD_CG_SUPPORT_GFX_CGCG |
+			/* Enabling CGCG on 4.9 kernel results in
+			 * IOMMU errors and filesystem corruption
+			 * Disable CGCG until the issue is fixed properly
+			 */
+			// AMD_CG_SUPPORT_GFX_CGCG |
 			AMD_CG_SUPPORT_GFX_CGLS |
 			AMD_CG_SUPPORT_BIF_LS |
 			AMD_CG_SUPPORT_HDP_MGCG |
-- 
2.7.4

